src/shell/oom.c

```diff
@@ -162,6 +163,80 @@ static struct shell_oom *oom_create (flux_shell_t *shell, flux_error_t *errp)
     return NULL;
 }
 
+static int peek_int (pid_t pid, const char *name, int *val)
+{
+    char path[1024];
+    FILE *fp;
+    int value;
+
+    snprintf (path, sizeof (path), "/proc/%d/%s", pid, name);
+    if (!(fp = fopen (path, "r")))
+        return -1;
+    if (fscanf (fp, "%d", &value) != 1) {
+        ERRNO_SAFE_WRAP (fclose, fp);
+        return -1;
+    }
+    (void)fclose (fp);
+    *val = value;
+    return 0;
+}
+
+static int poke_int (pid_t pid, const char *name, int val)
+{
+    char path[1024];
+    FILE *fp;
+
+    snprintf (path, sizeof (path), "/proc/%d/%s", pid, name);
+    if (!(fp = fopen (path, "w")))
+        return -1;
+    if (fprintf (fp, "%d", val) < 0) {
+        ERRNO_SAFE_WRAP (fclose, fp);
+        return -1;
+    }
+    if (fclose (fp) < 0)
+        return -1;
+    return 0;
+}
+
+static int oom_adjust (flux_plugin_t *p,
+                       const char *topic,
+                       flux_plugin_arg_t *args,
+                       void *data)
+{
+    int adjust = ptr2int (data);
+    flux_shell_t *shell = flux_plugin_get_shell (p);
+    flux_shell_task_t *task;
+    pid_t pid;
+    int old_adjust;
+    int score;
+    bool failed = false;
+    int saved_errno;
+
+    if (!(shell = flux_plugin_get_shell (p))
+        || !(task = flux_shell_current_task (shell))
+        || flux_shell_task_info_unpack (task, "{s:I}", "pid", &pid) < 0)
+        return shell_log_errno ("failed to get current task pid");
+
+    if (peek_int (pid, "oom_score_adj", &old_adjust) < 0)
+        return -1;
+    if (poke_int (pid, "oom_score_adj", adjust) < 0) {
+        saved_errno = errno;
+        failed = true;
+    }
```

Comment:
If writing the new value doesn't cause the job to fail, should failure reading it cause the job to fail?

Reply:
Hmm yeah, maybe not a great idea to always expect those values to be readable.  Will fix.